lib/rand.c

```diff
@@ -100,7 +100,7 @@ CURLcode Curl_win32_random(unsigned char *entropy, size_t length)
 }
 #endif
 
-#if !defined(USE_SSL) || defined(USE_RUSTLS)
```

Comment:
I notice some CI runs are now failing the "single use function check" ([exemplar](https://github.com/cpu/curl/actions/runs/10836519408/job/30070467574)).

I could use some advice on how best to fix that one. It seems like the `!defined(USE_SSL)` build configuration would still want this function to exist, so it shouldn't be deleted outright (?). Is the intended fix to in-line this somewhere?

Reply:
Hm, weird. I'll have a look.

Reply:
Ah, not so weird really. The function is now *not* static and uses a `Curl_` prefix, but there is nothing outside the file that uses it so it should rather be made static (and not use the global prefix). Like this:

~~~diff
diff --git a/lib/rand.c b/lib/rand.c
index 3e4f500c0..edc293d6f 100644
--- a/lib/rand.c
+++ b/lib/rand.c
@@ -100,13 +100,13 @@ CURLcode Curl_win32_random(unsigned char *entropy, size_t length)
 }
 #endif
 
 #if !defined(USE_SSL)
 /* ---- possibly non-cryptographic version following ---- */
-CURLcode Curl_weak_random(struct Curl_easy *data,
-                          unsigned char *entropy,
-                          size_t length) /* always 4, size of int */
+static CURLcode weak_random(struct Curl_easy *data,
+                            unsigned char *entropy,
+                            size_t length) /* always 4, size of int */
 {
   unsigned int r;
   DEBUGASSERT(length == sizeof(int));
 
   /* Trying cryptographically secure functions first */
@@ -149,11 +149,11 @@ CURLcode Curl_weak_random(struct Curl_easy *data,
 #endif
 
 #ifdef USE_SSL
 #define _random(x,y,z) Curl_ssl_random(x,y,z)
 #else
-#define _random(x,y,z) Curl_weak_random(x,y,z)
+#define _random(x,y,z) weak_random(x,y,z)
 #endif
 
 static CURLcode randit(struct Curl_easy *data, unsigned int *rnd,
                        bool env_override)
 {

Reply:
:bulb: Aha! That makes sense, thank you. I've applied your suggested diff.